src/group_theory/nilpotent.lean

```diff
@@ -549,6 +576,102 @@ instance nilpotent_quotient_of_nilpotent (H : subgroup G) [H.normal] [h : is_nil
 lemma nilpotency_class_quotient_le (H : subgroup G) [H.normal] [h : is_nilpotent G] :
   group.nilpotency_class (G ⧸ H) ≤ group.nilpotency_class G := nilpotency_class_le_of_surjective _ _
 
+lemma comap_comap_center {H₁ : subgroup G} [H₁.normal] {H₂ : subgroup (G ⧸ H₁)} [H₂.normal] :
+  comap (mk' H₁) (comap (mk' H₂) (center ((G ⧸ H₁) ⧸ H₂))) =
+    comap (mk' (comap (mk' H₁) H₂)) (center (G ⧸ comap (mk' H₁) H₂)) :=
+begin
+  ext x,
+  simp only [mk'_apply, mem_comap, mem_center_iff, forall_coe],
+  apply forall_congr,
+  change ∀ (y : G), (↑↑(y * x) = ↑↑(x * y) ↔ ↑(y * x) = ↑(x * y)),
+  intro y,
+  repeat { rw [eq_iff_div_mem] },
+  simp,
+end
+
+-- This lemma helps with rewriting the subgroup, which occurs in indices
+lemma comap_center_subst {H₁ H₂ : subgroup G} [normal H₁] [normal H₂] (h : H₁ = H₂) :
+  comap (mk' H₁) (center (G ⧸ H₁)) = comap (mk' H₂) (center (G ⧸ H₂)) :=
+  by { unfreezingI { subst h, } }
```

Comment:
These two lemmas are technical and rather specific for the following proof. Should I mark them as `private` maybe?

Reply:
The previous lemma seems technical, but fine as a public lemma. This lemma is indeed very specific. Ideally it would be possible to avoid it.
Aside: Note that you can replace `by { tac }` with `by tac` if the proof is a single tactic application.

Reply:
> The previous lemma seems technical, but fine as a public lemma

I assume I should move it to `subgroup/quotient_group` then?

> This lemma is indeed very specific. Ideally it would be possible to avoid it.

Ideally, it would just be `rw h` at the place where I currently need the lemma, but `rw` doesn’t seem to be able to rewrite in terms that are also indices elsewhere (it becomes ill-typed), and I don’t know of an equivalent of Coq’s `dependent_rewrite`. See [discussion on zuplip](https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/dependent.20rewrite). If you have a suggestion about how to avoid it, I’d be very curious.

Reply:
Yes, please move the other lemma to the file on quotient groups. This other specific lemma is probably fine as a private lemma, as long as we don't have a version of `dependent_rewrite`.